presto-main/src/main/java/com/facebook/presto/sql/planner/RelationPlanner.java

```diff
@@ -243,24 +238,6 @@ protected RelationPlan visitJoin(Join node, Void context)
             for (Expression conjunct : ExpressionUtils.extractConjuncts(criteria)) {
                 conjunct = ExpressionUtils.normalize(conjunct);
 
-                // We skip this section for joins using "USING" clause as generated criteria uses non qualified column names.
```

Comment:
I think the correct approach here is to drop the entire expression into the JoinNode filterCondition and let the predicate push down extract out the things that are needed. I think this will work with your changes below and one additional change.